source/nodejs/adaptivecards/src/serialization.ts

```diff
@@ -1086,10 +1086,23 @@ export abstract class SerializableObject {
         }
     }
 
-    protected getValue(prop: PropertyDefinition): any {
-        return this._propertyBag.hasOwnProperty(prop.getInternalName())
-            ? this._propertyBag[prop.getInternalName()]
-            : prop.defaultValue;
+    protected getValue(prop: PropertyDefinition, overrideProperty: boolean = false): any {
+        const name = prop.getInternalName();
+        if (this._propertyBag.hasOwnProperty(name)) {
+            const currentValue = this._propertyBag[name];
+
+            if (overrideProperty && this._rawProperties.hasOwnProperty(name)) {
```

Comment:
can you change overrideProperty to convey its boolean type?

for example, it can be shouldOverrideProperty, it makes the meaning much clear.